File: src/main/java/me/aov/sellgui/EssentialsHolder.java
//
// Source code recreated from a .class file by IntelliJ IDEA
// (powered by FernFlower decompiler)
//

package me.aov.sellgui;

import com.earth2me.essentials.Essentials;
import java.math.BigDecimal;
import org.bukkit.Bukkit;
import org.bukkit.inventory.ItemStack;

public class EssentialsHolder {
    private Essentials essentials = (Essentials)Bukkit.getPluginManager().getPlugin("Essentials");

    public EssentialsHolder() {
        if (this.essentials == null) {
            System.out.println("WEEWOOO");
        }

    }

    public Essentials getEssentials() {
        return this.essentials;
    }

    public void setEssentials(Essentials essentials) {
        this.essentials = essentials;
    }

    public BigDecimal getPrice(ItemStack itemStack) {
        return this.essentials.getWorth().getPrice(this.essentials, itemStack) != null ? this.essentials.getWorth().getPrice(this.essentials, itemStack) : new BigDecimal(0);
    }
}

File: src/main/kotlin/me/aov/sellgui/EssentialsHolder.kt
package me.aov.sellgui

import com.earth2me.essentials.Essentials
import org.bukkit.Bukkit
import org.bukkit.inventory.ItemStack
import java.math.BigDecimal

class EssentialsHolder {
    private var essentials: Essentials? = Bukkit.getPluginManager().getPlugin("Essentials") as? Essentials

    init {
        if (essentials == null) {
            println("WEEWOOO")
        }
    }

    fun getEssentials(): Essentials? = essentials

    fun setEssentials(essentials: Essentials?) {
        this.essentials = essentials
    }

    fun getPrice(itemStack: ItemStack): BigDecimal {
        val currentEssentials = essentials ?: return BigDecimal.ZERO
        return currentEssentials.worth.getPrice(currentEssentials, itemStack) ?: BigDecimal.ZERO
    }
}
